src/project/crate_data.rs

```diff
@@ -20,6 +20,24 @@ use itertools::chain;
 use super::builtin_plugins::BuiltinPlugin;
 use crate::TRICKS;
 use crate::lang::db::AnalysisDatabase;
+use crate::project::model::MemberConfig;
+
+#[derive(Debug)]
+pub struct CrateInfo {
+    pub cr: Crate,
+    pub tools_config: Option<MemberConfig>,
+    /// Path to Scarb.toml.
+    pub manifest_path: Option<PathBuf>,
```

Comment:
```suggestion
    pub tools_config: MemberConfig,
    /// Path to Scarb.toml.
    pub manifest_path: PathBuf,
```

There is no case where it can be `None` I believe.